src/smart-components/Channel/components/MessageList/index.tsx

```diff
@@ -114,73 +121,94 @@ const MessageList: React.FC<MessageListProps> = (props: MessageListProps) => {
     }
   };
 
-  // Because every message components are re-rendered everytime by every scroll events
-  const memoizedAllMessages = useMemo(() => {
-    return (
-      allMessages.map((m, idx) => {
-        const previousMessage = allMessages[idx - 1];
-        const nextMessage = allMessages[idx + 1];
-        const [chainTop, chainBottom] = isMessageGroupingEnabled
-          ? compareMessagesForGrouping(previousMessage, m, nextMessage, currentGroupChannel, replyType)
-          : [false, false];
-        const previousMessageCreatedAt = previousMessage?.createdAt;
-        const currentCreatedAt = m.createdAt;
-        // https://stackoverflow.com/a/41855608
-        const hasSeparator = !(previousMessageCreatedAt && (
-          isSameDay(currentCreatedAt, previousMessageCreatedAt)
-        ));
-
-        const handleScroll = () => {
-          const current = scrollRef?.current;
-          if (current) {
-            const bottom = current.scrollHeight - current.scrollTop - current.offsetHeight;
-            if (scrollBottom < bottom) {
-              current.scrollTop += bottom - scrollBottom;
-            }
-          }
-        };
-
-        return (
-          <Message
-            key={m?.messageId}
-            handleScroll={handleScroll}
-            renderMessage={renderMessage}
-            message={m}
-            hasSeparator={hasSeparator}
-            chainTop={chainTop}
-            chainBottom={chainBottom}
-            renderCustomSeparator={renderCustomSeparator}
-          />
-        );
-      })
-    );
-  }, [allMessages]);
+  const handleScroll = () => {
+    const current = scrollRef?.current;
+    if (current) {
+      const bottom = current.scrollHeight - current.scrollTop - current.offsetHeight;
+      if (scrollBottom < bottom && scrollBottom <= MESSAGE_SCROLL_BUFFER) {
+        current.scrollTop += bottom - scrollBottom;
+      }
+    }
+  };
 
+  if (loading) {
+    if (renderPlaceholderLoader && typeof renderPlaceholderLoader === 'function') {
```

Comment:
nit, can simplify this -> (loading && (typeof renderPlaceholderLoader === 'function'))